react-components/src/query/useAll3dDirectConnectionsWithProperties.ts

```diff
@@ -145,27 +144,26 @@ export function useAll3dDirectConnectionsWithProperties(
           )
           .filter((item) => item.items.length > 0);
 
-      const instanceWithData =
-        instanceItemsAndTyping?.flatMap((itemsData) => {
-          let connectionFound: FdmConnectionWithNode | undefined;
-
-          itemsData.items.every((itemData) => {
+      const instanceWithData = instanceItemsAndTyping.flatMap((itemsData) => {
+        const dataFound = itemsData.items
+          .map((itemData) => {
             const fdmKey = createFdmKey(itemData);
-            if (connectionWithNodeAndViewMap.has(fdmKey)) {
-              connectionFound = connectionWithNodeAndViewMap.get(fdmKey);
-              return false;
-            }
-            return true;
-          });
 
-          if (connectionFound === undefined) return [];
+            if (!connectionWithNodeAndViewMap.has(fdmKey)) return undefined;
```

Comment:
Suggestion:

Since we're within a `flatMap` iteration, returning `[]` will flatten to nothing, meaning the calls to `flat` and `filter(isDefined)` afterwards would be redundant (isn't `flat` already redundant?).